packages/sanity/src/core/store/_legacy/document/document-pair/memoizedPair.ts

```diff
@@ -15,5 +15,11 @@ export const memoizedPair: (
       subscriber.next(checkoutPair(client, idPair))
     }).pipe(publishReplay(1), refCount())
   },
-  (_client, idPair) => idPair.publishedId
+  (client: SanityClient, idPair: IdPair, typeName?: string) => {
+    const config = client.config()
+
+    return `${config.dataset ?? ''}-${config.projectId ?? ''}-${idPair.publishedId}-${
```

Comment:
Passing a client without dataset/projectId here will probably cause problems down the line, so I'm wondering if it makes sense to throw an error here instead if that's the case. WDYT?